src/crystal/system/file.cr

```diff
@@ -1,8 +1,51 @@
+require "file/mode"
+
 # :nodoc:
 module Crystal::System::File
   # Helper method for calculating file open modes on systems with posix-y `open`
   # calls.
-  private def self.open_flag(mode)
+  private def self.open_flag(mode : ::File::Mode)
+    if mode.overwrite?
+      mode |= ::File::Mode::Write | ::File::Mode::Create | ::File::Mode::Truncate
+      mode &= ~(::File::Mode::Overwrite)
+    end
+
+    flags = 0
+    if mode.read? && (mode.write? || mode.append?)
+      flags |= LibC::O_RDWR
+      flags |= LibC::O_APPEND if mode.append?
+      mode &= ~(::File::Mode::Read | ::File::Mode::Write | ::File::Mode::Append)
+    end
+    mode.each do |m|
+      case m
+      when ::File::Mode::Read      then flags |= LibC::O_RDONLY
+      when ::File::Mode::Write     then flags |= LibC::O_WRONLY
+      when ::File::Mode::Create    then flags |= LibC::O_CREAT
+      when ::File::Mode::CreateNew then flags |= LibC::O_CREAT | LibC::O_EXCL
+      when ::File::Mode::Append    then flags |= LibC::O_WRONLY | LibC::O_APPEND
+      when ::File::Mode::Truncate  then flags |= LibC::O_TRUNC
+      when ::File::Mode::Sync
+        {% if flag?(:win32) %}
+          raise NotImplementedError.new("File.open(..., :sync)")
+        {% else %}
+          flags |= LibC::O_SYNC
+        {% end %}
+      when ::File::Mode::SymlinkNoFollow
+        {% if flag?(:win32) %}
+          raise NotImplementedError.new("File.open(..., :symlink_no_follow)")
+        {% else %}
+          flags |= LibC::O_NOFOLLOW
+        {% end %}
+      else
+        raise "Unknown mode #{m}"
+      end
+    end
+    flags
+  end
+
+  # Helper method for calculating file open modes on systems with posix-y `open`
+  # calls.
+  private def self.open_flag(mode : String)
```

Comment:
Your idea is to remove this eventually together with the deprecated methods, right?

Reply:
Yep.